### PR TITLE
Fix unhashable decay_var_list

### PR DESCRIPTION
**Pull request by theoldfather**
Fixes an error where the list of decay variables is unhashable because they are compared without var.ref() or var.experimental_ref(). This patch is forward compatible with TF 2.2.0.

### CODE REVIEW

File: tensorflow_addons/optimizers/weight_decay_optimizers.py [hunk 1]
```diff
@@ -21,6 +21,14 @@
 from typing import Union, Callable, Type
 
 
+def _ref(var):
```
Comment (PhilJd) [inline tensorflow_addons/optimizers/weight_decay_optimizers.py]: tf.addons supports TF2 only anyway, maybe we can remove this string comparison and just return var.ref()/experimental_ref()?

**Comment by gabrieldemarmiesse** [comment]
@theoldfather could you provide a test in your pull request to ensure that we don't re-introduce the bug in the future?

Ideally this test should fail without your patch.

File: tensorflow_addons/optimizers/weight_decay_optimizers.py (hunk 1 above)
Comment (theoldfather) [reply tensorflow_addons/optimizers/weight_decay_optimizers.py]: Done.

**Comment by theoldfather** [comment]
> @theoldfather could you provide a test in your pull request to ensure that we don't re-introduce the bug in the future?
> 
> Ideally this test should fail without your patch.

Done.

**Comment by gabrieldemarmiesse** [comment]
@theoldfather  thank you! I'll take a look.

**Comment by theoldfather** [comment]
Thanks for reviewing!

**Pull request merged**